internal/testbed/integration/filtering_test.go

```diff
@@ -96,6 +96,209 @@ func TestFilteringCreditCard(t *testing.T) {
+	attributesMasked.PutStr("iban26", "****")
+	attributesMasked.PutStr("iban27", "****")
+	attributesMasked.PutStr("iban28", "****")
+	attributesMasked.PutStr("iban29", "****")
+	attributesMasked.PutStr("iban30", "****")
+	attributesMasked.PutStr("non-iban", "no4444 ds")
+	attributesMasked.PutInt("redaction.masked.count", int64(30))
+
+	attributesFiltered := pcommon.NewMap()
+	attributesFiltered.PutStr("iban1", "DE **** 30 00")
+	attributesFiltered.PutStr("iban2", "FR **** 2 606")
+	attributesFiltered.PutStr("iban3", "ES **** 1332")
+	attributesFiltered.PutStr("iban4", "IT **** 3 456")
+	attributesFiltered.PutStr("iban5", "NL **** 43 00")
+	attributesFiltered.PutStr("iban6", "BE **** 6769")
+	attributesFiltered.PutStr("iban7", "AT **** 4568")
+	attributesFiltered.PutStr("iban8", "SE **** 3242")
+	attributesFiltered.PutStr("iban9", "PL **** 2874")
+	attributesFiltered.PutStr("iban10", "GB **** 68 19")
+	attributesFiltered.PutStr("iban11", "AL **** 8741")
+	attributesFiltered.PutStr("iban12", "CY **** 7600")
+	attributesFiltered.PutStr("iban13", "KW **** 01 01")
+	attributesFiltered.PutStr("iban14", "LU **** 0000")
+	attributesFiltered.PutStr("iban15", "NO **** 7 947")
+	attributesFiltered.PutStr("iban16", "DE **** 3000")
+	attributesFiltered.PutStr("iban17", "FR **** 2606")
+	attributesFiltered.PutStr("iban18", "ES **** 1332")
+	attributesFiltered.PutStr("iban19", "IT **** 3456")
+	attributesFiltered.PutStr("iban20", "NL **** 4300")
+	attributesFiltered.PutStr("iban21", "BE **** 6769")
+	attributesFiltered.PutStr("iban22", "AT **** 4568")
+	attributesFiltered.PutStr("iban23", "SE **** 3242")
+	attributesFiltered.PutStr("iban24", "PL **** 2874")
+	attributesFiltered.PutStr("iban25", "GB **** 6819")
+	attributesFiltered.PutStr("iban26", "AL **** 8741")
+	attributesFiltered.PutStr("iban27", "CY **** 7600")
+	attributesFiltered.PutStr("iban28", "KW **** 0101")
+	attributesFiltered.PutStr("iban29", "LU **** 0000")
+	attributesFiltered.PutStr("iban30", "NO **** 7947")
+	attributesFiltered.PutStr("non-iban", "no4444 ds")
+
+	ibanRedactionProcessor := map[string]string{
+		"redaction": `
+  redaction:
+    allow_all_keys: true
+    blocked_values:
+      - "^[A-Z]{2}[0-9]{2}(\\s*[A-Z0-9]){8,30}$"
+    summary: info
+`,
+	}
+
+	ibanTransformProcessor := map[string]string{
+		"transform": `
+  transform:
+    error_mode: ignore
+    trace_statements:
+      - context: span
+        statements:
+          - replace_all_patterns(attributes, "value", "^([A-Z]{2})[0-9]{2}((\\s*[A-Z0-9]){4,26})\\s*((\\s*[A-Z0-9]){4})$", "$$1 **** $$4")
+    metric_statements:
+      - context: datapoint
+        statements:
+          - replace_all_patterns(attributes, "value", "^([A-Z]{2})[0-9]{2}((\\s*[A-Z0-9]){4,26})\\s*((\\s*[A-Z0-9]){4})$", "$$1 **** $$4")
+    log_statements:
+      - context: log
+        statements:
+          - replace_all_patterns(attributes, "value", "^([A-Z]{2})[0-9]{2}((\\s*[A-Z0-9]){4,26})\\s*((\\s*[A-Z0-9]){4})$", "$$1 **** $$4")
+`,
+	}
```

Comment:
Can this be taken directly from `config_examples\masking_iban.yaml` rather than duplicating and inlining it?

This way, it would always be kept in sync and changes to the YAML file would be tested directly while we wouldn't need to think of reflecting changes here as well.

Also, the invalid config pointed out in https://github.com/Dynatrace/dynatrace-otel-collector/pull/343#discussion_r1814893699 would've been spotted immediately as the test would've failed.

Reply:
we should be able to extract only the processors part from the config to avoid rewriting the whole filtering scenario, since it's already used in multiple tests (most of them non-merged, but ready). I will try to create a wrapper, that will do the extraction from the example config to a map, since in our cases, receiver/exporters are not important here. Do you agree?

Reply:
adapted the functionality [here](https://github.com/Dynatrace/dynatrace-otel-collector/pull/340/commits/000067fbdb38c89f339187c83cba6a6abb12d6a4) as part of a different PR (that should be merged before this one), please have a look there if you have any objections, otherwise I will proceed with merging

Reply:
That change looks good to me. Thanks!